### PR TITLE
[Serve][Doc] Moves Serve REST API and Serve CLI API into separate subpages

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Restructuring docs. Moves [Serve REST API](https://docs.ray.io/en/master/serve/package-ref.html#serve-rest-api) and [Serve CLI API](https://docs.ray.io/en/master/serve/package-ref.html#serve-cli) into separate subpages.

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/package-ref.md [hunk 1]
```diff
@@ -47,169 +47,11 @@
 .. autofunction:: ray.serve.batch(max_batch_size=10, batch_wait_timeout_s=0.0)
 ```
 
-## Serve REST API
+## Extended API
```
Comment (zcin) [inline doc/source/serve/package-ref.md:50]: Are there any suggestions for another name for this section? "Non-Python API" doesn't sound as nice, "Other API" is possible although generic. cc: @shrekris-anyscale @simon-mo @edoakes

Comment (simon-mo) [reply doc/source/serve/package-ref.md:50]: Operational APIs
Please checkout the REST API and CLI that are useful for running, debugging, inspecting, and deploying the application in production.

**Review by simon-mo — APPROVED** [review]
@shrekris-anyscale can you help add a sentence or two the description of both REST and CLI page

**Review by simon-mo — APPROVED** [review]

**Comment by simon-mo** [comment]
Please fix the merge conflict, and we can get this in

**Pull request merged**